### PR TITLE
Add disable_gc opt-out for GC-free attach workloads

### PR DESCRIPTION
**Pull request by hackerwins**
## Summary

Adds a per-request `disable_gc` flag so a client can opt out of receiving the response `VersionVector` and participating in server-side `minVV` tracking. Targets high-fan-out Counter / primitive workloads where no client consumes tombstones, eliminating both the per-PushPull `UpdateMinVersionVector` compute and the per-client `versionvectors` row.

- Adds `bool disable_gc` to **both** `AttachDocumentRequest` and `PushPullChangesRequest`. These are the two RPCs that drive a PushPull on behalf of an end client; `DetachDocument` and `RemoveDocument` deliberately do not carry the flag (one final minVV write at terminus is negligible).
- `packs.PushPullOptions.DisableGC` carries the flag from each handler into `pushpull.go`, which skips `UpdateMinVersionVector` and nils the response VV (both the change-info and snapshot pull paths).
- Adds `client.WithDisableGC()` attach option. The SDK records `disableGC` on the local `Attachment` and sends the matching wire field on every subsequent `PushPullChanges`.
- **Not persisted server-side.** Each PushPull reads the flag from the request alone. Persistence on `ClientDocInfo` was considered (centralised, set-once) but the per-request wire form is genuinely smaller end-to-end — no BSON field forever, no signature change on `AttachDocument` cascading through ~30 test call sites, no DeepCopy or cache-equality updates.
- Design: `docs/design/disable-gc-on-attach.md`

Spec covers Goals/Non-Goals, snapshot GC edge case (aggressive GC when all clients opt out — falls out of existing `MinVersionVector` semantics with no extra code), and deferred client-side optimizations (request-VV omission and skip local GC traversal are future work; the server-side wins are the goal of this PR).

JS SDK support is a follow-up PR after this Go side merges.

## Test plan

- [x] `make lint` clean
- [x] `go test ./...` — 1149 unit tests pass
- [x] `go test -tags integration ./test/integration/` — full integration suite green
- [x] `TestDisableGCOnAttach` covers: opt-out client writes no `versionvectors` row, mixed opt-in/opt-out coexistence converges, re-attach without the option restores normal behavior
- [x] Two passes of self-review via subagent; Important findings addressed in each iteration

## Summary by CodeRabbit

* **New Features**
  * Optional "disable GC" attach mode: clients can opt out of garbage-collection participation; syncs for opted-out attachments omit version-vector exchange and skip server min-VV tracking.

* **Documentation**
  * Added design and task docs detailing the opt-out contract, risks, and implementation notes.

* **Tests**
  * Added integration test validating opt-out attach, mixed-mode convergence, and re-attaching to resume GC tracking.

[Review Change Stack](https://app.coderabbit.ai/change-stack/yorkie-team/yorkie/pull/1822?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: docs/design/disable-gc-on-attach.md [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+        resPack.VersionVector = minVersionVector
+    }
+}
+```
+
+The flag is **not persisted** on `ClientDocInfo` or anywhere else on the
+server. Each PushPull reads it from the request alone. The cost is one
+bool per request; the savings are no schema field, no migration, no
+cache invalidation logic, and no risk of persisted state drifting out
+of sync with client intent.
```
[the remark this answers is not in the document]
Comment (hackerwins) [reply docs/design/disable-gc-on-attach.md:145]: The doc and the implementation are aligned in the current branch — neither persists `disable_gc`. The flag travels on `AttachDocumentRequest` and `PushPullChangesRequest` and is read by the handler into `PushPullOptions`. There is no `ClientDocInfo.DisableGC` field, no `$set` for `disable_gc` in mongo, no DeepCopy update.

The PR description originally described a persisted design; mid-implementation it pivoted to per-request after a review pass that pointed out the touch-point count favors stateless. The walkthrough above and any inline review summary that still references persistence are stale relative to the shipped code. The lessons file records the pivot rationale.

Closing as not actionable.
